util/ranger/ranger.go

```diff
@@ -256,32 +260,30 @@ func BuildColumnRange(conds []expression.Expression, sc *stmtctx.StatementContex
 	return ranges, nil
 }
 
-// BuildIndexRange builds the range for index.
-func BuildIndexRange(sc *stmtctx.StatementContext, cols []*expression.Column, lengths []int,
-	accessCondition []expression.Expression) ([]*NewRange, error) {
+// buildCNFIndexRange builds the range for index where the top layer is CNF.
+func buildCNFIndexRange(sc *stmtctx.StatementContext, cols []*expression.Column, newTp []*types.FieldType, lengths []int,
+	eqAndInCount int, accessCondition []expression.Expression) ([]*NewRange, error) {
 	rb := builder{sc: sc}
 	var (
-		ranges       []*NewRange
-		eqAndInCount int
-		err          error
+		ranges []*NewRange
+		err    error
 	)
-	newTp := make([]*types.FieldType, 0, len(cols))
```

Comment:
Why is newTp not constructed here ?

Reply:
There is not the entry point bow.  It's DetachCondAndBuildRangeForIndex.